python/cudf/cudf/core/reshape.py

```diff
@@ -1160,3 +1160,257 @@ def _length_check_params(obj, columns, name):
                 f"length of the columns being "
                 f"encoded ({len(columns)})."
             )
+
+
+def _get_names(arrs, names, prefix: str = "row"):
+    if names is None:
+        names = []
+        for i, arr in enumerate(arrs):
+            if isinstance(arr, cudf.Series) and arr.name is not None:
+                names.append(arr.name)
+            else:
+                names.append(f"{prefix}_{i}")
+    else:
+        if len(names) != len(arrs):
+            raise AssertionError("arrays and names must have the same length")
+        if not isinstance(names, list):
+            names = list(names)
+
+    return names
+
+
+def crosstab(
+    index,
+    columns,
+    values=None,
+    rownames=None,
+    colnames=None,
+    aggfunc=None,
+    margins=None,
+    margins_name=None,
+    dropna=None,
+    normalize=None,
+):
+    """
+    Compute a simple cross tabulation of two (or more) factors. By default
+    computes a frequency table of the factors unless an array of values and an
+    aggregation function are passed.
+
+    Parameters
```

Comment:
For any parameters that we don't support, can we simply document those as being unsupported? For example:


```

```
margins : Not supported
```

Reply:
I've already documented such parameters as **unsupported** but along with their description. Let me know if we don't want to keep the description. I thought it would be better if we know what's the use of those parameters. For example:-

```
margins : bool, default False
        margins is not supported yet
        Add row/column margins (subtotals).
```

Reply:
Yes - I'd rather we remove the description now and only add it back in when the parameter is supported. Otherwise, there's room for ambiguity.

Reply:
Done.